packages/block-library/src/site-logo/edit.js

```diff
@@ -0,0 +1,375 @@
+					left: showLeftHandle,
+				} }
+				onResizeStart={ onResizeStart }
+				onResizeStop={ ( event, direction, elt, delta ) => {
+					onResizeStop();
+					setAttributes( {
+						width: parseInt( currentWidth + delta.width, 10 ),
+						height: parseInt( currentHeight + delta.height, 10 ),
+					} );
+				} }
+			>
+				{ img }
+			</ResizableBox>
+		);
+
+		logoInspectorControls = (
+			<InspectorControls>
+				<PanelBody title={ __( 'Site Logo Settings' ) }>
+					<RangeControl
+						label={ __( 'Image width' ) }
+						onChange={ ( newWidth ) =>
+							setAttributes( { width: newWidth } )
+						}
+						min={ minWidth }
+						max={ maxWidthBuffer }
+						initialPosition={ Math.min(
+							naturalWidth,
+							maxWidthBuffer
+						) }
+						value={ width || '' }
+						disabled={ ! isResizable }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+	}
+
+	return (
+		<>
+			{ logoInspectorControls }
+			{ img }
+		</>
+	);
+};
+
+export default function LogoEdit( {
+	attributes,
+	className,
+	setAttributes,
+	isSelected,
+} ) {
+	const { width } = attributes;
+	const [ url, setUrl ] = useState( null );
+	const [ error, setError ] = useState();
+	const ref = useRef();
+	const { sitelogo } = useSelect( ( select ) =>
+		select( 'core' ).getEditedEntityRecord( 'root', 'site' )
+	);
+	const mediaItemData = useSelect(
+		( select ) => {
+			const mediaItem = select( 'core' ).getEntityRecord(
+				'root',
+				'media',
+				sitelogo
+			);
+			return (
+				mediaItem && {
+					url: mediaItem.source_url,
+					alt: mediaItem.alt_text,
+				}
+			);
+		},
+		[ sitelogo ]
+	);
+
+	const { editEntityRecord } = useDispatch( 'core' );
+	const setLogo = ( newValue ) =>
+		editEntityRecord( 'root', 'site', undefined, {
+			sitelogo: newValue,
+		} );
+
+	let alt = null;
+	if ( mediaItemData ) {
+		alt = mediaItemData.alt;
+		if ( url !== mediaItemData.url ) {
+			setUrl( mediaItemData.url );
+		}
+	}
+
+	const onSelectLogo = ( media ) => {
+		if ( ! media ) {
+			return;
+		}
+
+		if ( ! media.id && media.url ) {
+			// This is a temporary blob image
+			setLogo( '' );
+			setError();
+			setUrl( media.url );
+			return;
+		}
+
+		setLogo( media.id.toString() );
+	};
+
+	const deleteLogo = () => {
+		setLogo( '' );
+		setUrl( '' );
+	};
+
+	const onUploadError = ( message ) => {
+		setError( message[ 2 ] ? message[ 2 ] : null );
+	};
+
+	const controls = (
+		<BlockControls>
+			<ToolbarGroup>
+				{ url && (
+					<MediaReplaceFlow
+						mediaURL={ url }
+						allowedTypes={ ALLOWED_MEDIA_TYPES }
+						accept={ ACCEPT_MEDIA_STRING }
+						onSelect={ onSelectLogo }
+						onError={ onUploadError }
+					/>
+				) }
+				{ !! url && (
+					<ToolbarButton
+						icon="trash"
+						onClick={ () => deleteLogo() }
+						label={ __( 'Delete Site Logo' ) }
+					/>
+				) }
+			</ToolbarGroup>
+		</BlockControls>
+	);
+
+	const label = __( 'Site Logo' );
+	let logoImage;
+	if ( url === null ) {
+		logoImage = <Spinner />;
```

Comment:
I've tested on a site without logo, and when inserting the block, the placeholder shows up with an infinite spinner.

<img width="624" alt="Screenshot 2020-06-23 at 18 40 02" src="https://user-images.githubusercontent.com/2070010/85436442-fb03e800-b580-11ea-82a9-0175faa38ab0.png">

If I add a logo and remove it, the spinner would (correctly) disappear.